utils/log-collector

```diff
@@ -0,0 +1,101 @@
+#!/bin/bash
+
+# This script leverages the AWS CLI to collect Elastic Stack logs for your
+# Buildkite Elastic CI Stack instances.
+#
+# It assumes you have the AWS CLI tool configured, with appropriate credentials
+# so it is recommended to use something like aws-vault: https://github.com/99designs/aws-vault.
+#
+# You will need to know the Instance ID and Stack name of the resources
+# you want to collect logs for.
+# 
+# Once you have collected the logs, please send them to support@buildkite.com
+# with a description of the issue you are seeing, and we'll be happy to assist!
+
+set -euo pipefail
+
+usage() {
+cat << EOF
+Usage: ./bk-log-collector -s <stack-name> -i <instance-id>
+Collect Buildkite Elastic CI stack logs from AWS CloudWatch
+-s      Auto Scaling Group Name, can be found at console.aws.amazon.com/ec2autoscaling/
+-i      Instance ID, the AWS instance ID for the instance you want to collect logs for
+-o      Output format for logs(optional flag), supported values are text (default value), json, table
+EOF
+}
+
+while getopts 'hs:i:o:' opt; do
+    case ${opt} in
+        s ) 
+            stackName=$OPTARG
+            ;;
+        i )
+            instanceId=$OPTARG
+            ;;
+        o )
+            outputFormat=$OPTARG
+            ;;
+        h )
+            usage
+            exit 0
+            ;;
+        \? )
+            echo "Invalid Option: -$OPTARG" 1>&2
+            exit 1
+            ;;
+        : )
+            echo "Invalid option: $OPTARG requires an arguemnt" 1>&2
+            ;;
+    esac
+done
+shift $((OPTIND -1))
+
+now=$(date +"%F")
+outputFormat="${outputFormat:=text}"
+
+# Create a temporary directory to store our logs
+pwd=$(pwd)
+dir=$(mktemp -d "$pwd"/buildkite-logs-"$now")
+
+echo "Collecting logs for instance $instanceId"
+
+# Collect the logs for the instance
+aws logs get-log-events --log-group-name /buildkite/buildkite-agent --log-stream-name "$instanceId" --output "$outputFormat" >> "$dir"/buildkite-agent-"$instanceId".log
+aws logs get-log-events --log-group-name /buildkite/system --log-stream-name "$instanceId" --output "$outputFormat" >> "$dir"/system-"$instanceId".log
+aws logs get-log-events --log-group-name /buildkite/lifecycled --log-stream-name "$instanceId" --output "$outputFormat" >> "$dir"/lifecycled-"$instanceId".log
+aws logs get-log-events --log-group-name /buildkite/docker-daemon --log-stream-name "$instanceId" --output "$outputFormat" >> "$dir"/docker-daemon-"$instanceId".log
```

Comment:
I like the simplicity/directness, and would avoid any unnecessary complexity that would make this script harder to read/understand, but might make it _easier_ to read/understand?

```bash
for name in buildkite-agent system lifecycled docker-daemon; do
  aws logs get-log-events \
    --log-group-name "/buildkite/$name" \
    --log-stream-name "$instanceId" \
    --output "$outputFormat" \
    >> "$dir/$name-$instanceId.log"
end
```